File: backend/src/main/java/be/technobel/parsemaster/validation/constraint/NotBlank.java
package be.technobel.parsemaster.validation.constraint;

import be.technobel.parsemaster.validation.validator.NotBlankValidator;
import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.PARAMETER, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = NotBlankValidator.class)
public @interface NotBlank {
  String field() default "field";
  String message() default "blank";
  Class<?>[] groups() default {};
  Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/be/technobel/playzone/pl/validation/constraints/NotBlank.kt
package be.technobel.playzone.pl.validation.constraints

import be.technobel.playzone.pl.validation.validators.NotBlankValidator
import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [NotBlankValidator::class])
annotation class NotBlank(
	val field: String = "field",
	val message: String = "not_blank",
	val groups: Array<KClass<*>> = [],
	val payload: Array<KClass<out Payload>> = []
)